src/bp/ui-studio/src/web/views/OneFlow/diagram/index.tsx

```diff
@@ -859,7 +831,8 @@ class Diagram extends Component<Props> {
   }
 
   render() {
-    const formType: string = this.state.editingNodeItem?.node?.type
+    const formType: string = this.state.editingNodeItem?.node?.nodeType || this.state.editingNodeItem?.node?.type
```

Comment:
Why are there two different types ? Can't we just have one ? Also, I see that `this.state.editingNodeItem?.node` is repeated a lot, maybe we could create `const node = this.state.editingNodeItem?.node` and use this one

Reply:
The nodeType is added since initializing the BlockModel changes the type of the nodes to `block` (it's not saved in the database). So the nodeType will contain the type of the node after initialization, but when you add a new node, nodeType will be undefined and type will contain the right type since we open the form right away